### PR TITLE
Updated Dockerfile to use a multi-stage build.

### PR DESCRIPTION
**Pull request by bgsandan**
**Summary:**

This pull request updates the Dockerfile for the project to improve the build artifacts and deployment speed of the built container.  This is my first PR so if it needs formatting differently please let me know :)

**Related issues:** 

#830 and #640 

**Summary of changes:**

- Switched to alpine linux base image for smaller image sizes
- Created a multi-stage build allowing final image to only contain production dependencies
- All webpack compilation occurs during the docker build phase rather than some occurring when the container is executed.  This drastically reduces the time-to-live for the deployed dashboard

**Performance Impact**

Whilst initial container build time is increased, the container size and launch time are decreased.  The refactored Dockerfile should benefit from layer-caching when used in an iterative development cycle

Metrics (collected from 'cold' on a clean [Play with Docker](https://labs.play-with-docker.com/) instance):

|                | Container Build Time | Container Launch Time | Container Size |
|----------------|----------------------|-----------------------|----------------|
| Current Master | 1m 47.608s            | 0m 25.552s             | 320MB          |
| This PR        | 2m 42.572s            | 0m 1.328s              | 74.6MB         |

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+
+# Copy production node_modules
+COPY --from=base /src/prod_node_modules /src/node_modules
+COPY --from=base /src/package*.json /src/
+
+# Copy compiled src dirs
+COPY --from=base /src/Parse-Dashboard/ /src/Parse-Dashboard/
+
+
+EXPOSE 4040
```
Comment (flovilmart) [inline Dockerfile]: 4040 is only the default port. Is it relevant if the users has a config that listens to another port?

Comment (bgsandan) [reply Dockerfile]: Yep, good spot!  I don't think there is much point a user changing the listening port when running in Docker seeing as they can map the container port to a port of their choice on the host system, but to support this use case (and people porting their existing config to Docker) I've removed the default 4040 from the Dockerfile

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
+WORKDIR /src
+
+# Copy production node_modules
+COPY --from=base /src/prod_node_modules /src/node_modules
+COPY --from=base /src/package*.json /src/
+
+# Copy compiled src dirs
+COPY --from=base /src/Parse-Dashboard/ /src/Parse-Dashboard/
+
+ENTRYPOINT ["npm", "run", "start"]
```
Comment (flovilmart) [inline Dockerfile]: npm start is enough, no need to use `run`.

Also, I believe it is better to use node directly than using npm to start the server so we should use:

`node ./Parse-Dashboard/index.js` for the ENTRYPOINT

Comment (bgsandan) [reply Dockerfile]: I've updated the entrypoint and everything seems to still be working nicely :)

**Comment by flovilmart** [comment]
Thanks for the Pull Request! This surely will make the dashboard image lighter!

**Pull request merged**